src/main/ruby/core/truffle/boot.rb

```diff
@@ -27,6 +27,8 @@ def self.check_syntax(source_or_file)
     1
   end
 
+  private_class_method :check_syntax
```

Comment:
The convention for `private` and friends (in our codebase) is to add them right after the `end` of the method they apply to, so it makes a logical group of what belongs together.
Alternatively `private def my_method` works too but here with `private_class_method` it would be very long and not so readable.